src/storage/src/client/connections.rs

```diff
@@ -134,6 +134,13 @@ impl ConnectionContext {
             secrets_reader,
         }
     }
+    /// Sets the `secrets_reader` field on `self`.
+    ///
+    /// Returns `Self`.
+    pub fn secrets_reader(mut self, secrets_reader: Arc<dyn SecretsReader>) -> ConnectionContext {
+        self.secrets_reader = secrets_reader;
+        self
+    }
```

Comment:
Ah! Can you instead delete the `Default` impl, and add one method like:

```
impl ConnectionContext
    fn for_tests(secrets_reader: Arc<dyn SecretsReader>) {
        // The rest of the old Default impl.
    }
}
```

I'm 95% sure that `Default` impl was only ever used for tests; probs best just to make the correct usage in tests explicit.

Reply:
Done